src/main/java/com/ibm/etcd/client/utils/PersistentLeaseKey.java

```diff
@@ -168,9 +168,10 @@ protected void putKey(long leaseId) {
                     .elseDo().put(putBld.setIgnoreValue(false).setValue(defaultValue)).get(getOp)
                     .async();
             fut = Futures.transform(txnFut, (Function<TxnResponse,Object>)tr
-                    -> rangeCache.offerUpdate(tr.getResponses(1).getResponseRange().getKvs(0), false));
+                    -> rangeCache.offerUpdate(tr.getResponses(1).getResponseRange().getKvs(0), false),
+                    MoreExecutors.directExecutor());
         }
-        if(!isDone()) fut = Futures.transform(fut, (Function<Object,Object>) r -> set(key));
+        if(!isDone()) fut = Futures.transform(fut, (Function<Object,Object>) r -> set(key), MoreExecutors.directExecutor());
```

Comment:
cast can be removed; and maybe we can just have a static import for `directExecutor()`

Reply:
Done